test/libethereum/blockchain.cpp

```diff
@@ -757,48 +511,96 @@ mArray writeTransactionsToJson(Transactions const& _txs)
+	BOOST_CHECK_MESSAGE(blockHeaderFromFields.mixHash() == blockFromRlp.mixHash(), _testname + "mixHash in given RLP not matching the block mixHash!");
+	BOOST_CHECK_MESSAGE(blockHeaderFromFields.nonce() == blockFromRlp.nonce(), _testname + "nonce in given RLP not matching the block nonce!");
+
+	BOOST_CHECK_MESSAGE(blockHeaderFromFields == blockFromRlp, _testname + "However, blockHeaderFromFields != blockFromRlp!");
+
+	vector<TestTransaction> const& txsFromField = _blockFromFields.getTestTransactions();
+	vector<TestTransaction> const& txsFromRlp = _blockFromRlp.getTestTransactions();
+	BOOST_CHECK_MESSAGE(txsFromRlp.size() == txsFromField.size(), _testname + "transaction list size does not match");
+
+	for (size_t i = 0; i < txsFromField.size(); ++i)
+	{
+		Transaction const& trField = txsFromField.at(i).getTransaction();
+		Transaction const& trRlp = txsFromRlp.at(i).getTransaction();
+
+		BOOST_CHECK_MESSAGE(trField.data() == trRlp.data(), _testname + "transaction data in rlp and in field do not match");
+		BOOST_CHECK_MESSAGE(trField.gas() == trRlp.gas(), _testname + "transaction gasLimit in rlp and in field do not match");
+		BOOST_CHECK_MESSAGE(trField.gasPrice() == trRlp.gasPrice(), _testname + "transaction gasPrice in rlp and in field do not match");
+		BOOST_CHECK_MESSAGE(trField.nonce() == trRlp.nonce(), _testname + "transaction nonce in rlp and in field do not match");
+		BOOST_CHECK_MESSAGE(trField.signature().r == trRlp.signature().r, _testname + "transaction r in rlp and in field do not match");
+		BOOST_CHECK_MESSAGE(trField.signature().s == trRlp.signature().s, _testname + "transaction s in rlp and in field do not match");
+		BOOST_CHECK_MESSAGE(trField.signature().v == trRlp.signature().v, _testname + "transaction v in rlp and in field do not match");
+		BOOST_CHECK_MESSAGE(trField.receiveAddress() == trRlp.receiveAddress(), _testname + "transaction receiveAddress in rlp and in field do not match");
+		BOOST_CHECK_MESSAGE(trField.value() == trRlp.value(), _testname + "transaction receiveAddress in rlp and in field do not match");
+
+		BOOST_CHECK_MESSAGE(trField == trRlp, _testname + "transactions from  rlp and transaction from field do not match");
+		BOOST_CHECK_MESSAGE(trField.rlp() == trRlp.rlp(), _testname + "transactions rlp do not match");
+	}
 
-	return ret;
+	vector<TestBlock> const& unclesFromField = _blockFromFields.getUncles();
+	vector<TestBlock> const& unclesFromRlp = _blockFromRlp.getUncles();
+	BOOST_REQUIRE_EQUAL(unclesFromField.size(), unclesFromRlp.size());
+	for (size_t i = 0; i < unclesFromField.size(); ++i)
+		BOOST_CHECK_MESSAGE(unclesFromField.at(i).getBlockHeader() == unclesFromRlp.at(i).getBlockHeader(), _testname + "block header in rlp and in field do not match at uncles");
 }
 
-} }// Namespace Close
+}}//namespaces
 
 BOOST_AUTO_TEST_SUITE(BlockChainTests)
 
 BOOST_AUTO_TEST_CASE(bcForkBlockTest)
 {
-	if (!test::Options::get().fillTests)
```

Comment:
I did this to prevent errors when filling BlockChainTests which don't have a filler, how do you handle this situation now?